tests/conftest.py

```diff
@@ -243,12 +247,36 @@ def save_log_on_exit(request):
 
 
 @pytest.fixture(scope='module')
-def add_curl(request, init_session, setup_bootstrap):
+def add_curl(request, init_session):
     def _run_curl_pod():
-        if not setup_bootstrap.pods:
-            raise Exception("Could not find bootstrap node")
-
-        pod.create_pod(tests_conf.CURL_POD_FILE, testconfig['namespace'])
+        pod.create_pod(conf.CURL_POD_FILE, testconfig['namespace'])
         return True
 
     return _run_curl_pod()
+
+
+@pytest.fixture(scope='module')
+def add_node_pool():
+    """
+    memory should be represented by number of megabytes, \d*M
+
+    :return:
+    """
+    deployer = NodePoolDep(testconfig)
+    _, time_elapsed = deployer.add_node_pool()
+    yield time_elapsed
+    deployer.remove_node_pool()
```

Comment:
maybe add some indicator about time passed so we have some feeling of how much time we're waiting ?

Reply:
also wouldn't this code remove the node_pool right after its created?

Reply:
added a print for time_elapsed.
and about your question, no.
everything after the yield command is working as a destructor and will run only when this fixture will perform teardown.